opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -309,5 +304,28 @@ def get_aggregated_resources(
                 )
                 detected_resources = _EMPTY_RESOURCE
             finally:
-                final_resource = final_resource.merge(detected_resources)
-    return final_resource
+                detectors_resource = detectors_resource.merge(
+                    detected_resources
+                )
+
+    merged_resource = (
+        detectors_resource.merge(final_resource)
+        if final_resource
+        else detectors_resource
+    )
```

Comment:
I think this transition from `detectors_resource` -> `merged-resource` makes sense, but let me know if you have comments!

Or even just comments about the ternary statement to initialize them.